xbmc/xbmc.cpp

```diff
@@ -53,7 +53,7 @@ extern "C" int XBMC_Run(bool renderGUI)
   g_RBP.LogFirmwareVerison();
 #endif
 
-  if (renderGUI && !g_application.CreateGUI())
+  if (renderGUI && !g_application.IsHeadless() && !g_application.CreateGUI())
```

Comment:
I think it's used by people in the forum to do headless. So yeah, the
answer may be that we should drop it. They can use the library but do
application.setheadless(true) instead.

Another option is to make an options struct and send along to run? And make
AppParser create that struct. Might be nicer as application is hidden?